sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -504,16 +504,29 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
       private[this] var iter = computeInternal(split, context)
 
       override def hasNext: Boolean = try {
```

Comment:
This is to catch runtime code gen issue. We are not worried about actual compilation issues ? Should we also not dump code if it fails during compilation ?

Reply:
For compilation issues, the code is dumped even today in the lead.